components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -34,6 +34,9 @@ RewardsNotificationServiceImpl::RewardsNotificationServiceImpl(Profile* profile)
           std::make_unique<ExtensionRewardsNotificationServiceObserver>(
               profile);
   AddObserver(extension_rewards_notification_service_observer_.get());
+  for (auto& notification : rewards_notifications_) {
+    TriggerOnNotificationAdded(notification.second);
```

Comment:
I don't think we want to re-trigger for all observers, that might cause duplicates, but either way this isn't the right place to do this. We should be doing it in the AddObserver method. This should apply to all observers, not just the extension_rewards_notification_service_observer

Reply:
OK, I see now. The AddObserver method doesn't currently have access to the pending notifications, so I'll have to plumb those through. I'm thinking to just add a getter on the notification service, unless you have a better idea there.